cloud/auth/auth.go

```diff
@@ -369,9 +370,20 @@ func Login(domain string, client astro.Client, out io.Writer, loginLink bool) er
 
 	c, _ := context.GetCurrentContext()
 
-	res, err = authenticator.authDeviceLogin(c, authConfig, loginLink, domain)
-	if err != nil {
-		return err
+	if !tokenLogin {
+		res, err = authenticator.authDeviceLogin(c, authConfig, loginLink, domain)
+		if err != nil {
+			return err
+		}
+	} else {
```

Comment:
For more readability, you should return nil at the end of the if statement, it allows you to completely remove the else block

Reply:
If I add a return nill it will skip the rest of the function which is not wanted?